@here/harp-mapview/scripts/HarpWebpackConfig.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as CopyWebpackPlugin from "copy-webpack-plugin";
+import * as HtmlWebpackPlugin from "html-webpack-plugin";
+import { Configuration, Plugin } from "webpack";
+import * as WebpackMerge from "webpack-merge";
+
+export function addHarpWebpackConfig(
+    config?: Configuration,
+    mainPath?: string,
+    decoderPath?: string,
+    htmlPath?: string
+) {
+    const userConfig = config !== undefined ? config : {};
+    const mode = process.env.NODE_ENV === "production" ? "production" : "development";
+    const baseConfig: Configuration = {
+        output: {
+            filename: "[name].bundle.js"
+        },
+        devtool: "source-map",
+        resolve: {
+            extensions: [".webpack.js", ".web.js", ".js"]
+        },
+        performance: {
+            hints: false
+        },
+        mode
+    };
+    const typescriptConfig: Configuration = {
+        externals: [
+            {
+                three: "THREE"
+            },
+            (context: any, request: any, callback: (error?: any, result?: any) => void) => {
+                return /three\.module\.js$/.test(request) ? callback(null, "THREE") : callback();
+            }
+        ],
+        resolve: {
+            extensions: [".web.ts", ".ts", ".tsx"]
+        },
+        module: {
+            rules: [{ test: /\.tsx?$/, loader: "ts-loader" }]
+        }
+    };
+    const mainConfig =
+        mainPath !== undefined && mainPath.endsWith(".ts")
+            ? WebpackMerge.smart(baseConfig, typescriptConfig)
+            : baseConfig;
+    const bundles = [
+        WebpackMerge.smart(
+            {
+                ...mainConfig,
+                plugins: createPlugins(htmlPath),
+                stats: {
+                    all: false,
+                    timings: true,
+                    exclude: "/resources/",
+                    errors: true,
+                    entrypoints: true,
+                    warnings: true
+                }
+            },
+            userConfig
```

Comment:
Does it work if `userConfig` is already "array of bundles" ?

Reply:
Nope. I had a check for that before I switched to typescript, I will restore it so we throw an exception.
The user will need to handle merging themselves in that case.